From ed1725104e3e7145b4d7775f6f2bb510f692cd22 Mon Sep 17 00:00:00 2001
From: Binh Nguyen <binh.nguyen.uw@rvc.renesas.com>
Date: Mon, 14 Sep 2015 11:02:27 +0700
Subject: [PATCH] qtmultimedia: workaround for segmentation fault when stop
 video

wait for 200ms before stop qtvideorenderersink to ensure glTexSubimage2D() was finsish

Signed-off-by: Binh Nguyen <binh.nguyen.uw@rvc.renesas.com>
---
 .../src/gsttools/qgstvideorenderersink.cpp         |    4 ++++
 1 file changed, 4 insertions(+)

diff --git a/src/gsttools/qgstvideorenderersink.cpp b/src/gsttools/qgstvideorenderersink.cpp
index 31ac94e..f5807ad 100755
--- a/src/gsttools/qgstvideorenderersink.cpp
+++ b/src/gsttools/qgstvideorenderersink.cpp
@@ -48,6 +48,8 @@
 
 #include "qgstutils_p.h"
 
+#include <unistd.h> /*for usleep()*/
+
 //#define DEBUG_VIDEO_SURFACE_SINK
 
 QT_BEGIN_NAMESPACE
@@ -564,6 +566,8 @@ gboolean QGstVideoRendererSink::propose_allocation(GstBaseSink *base, GstQuery *
 
 gboolean QGstVideoRendererSink::stop(GstBaseSink *base)
 {
+    /*FIXME : Wait 200ms for finishing binding video frame*/
+    usleep(200000);
     VO_SINK(base);
     sink->delegate->stop();
     return TRUE;
-- 
1.7.9.5

